platform/darwin/src/MGLLight.h

```diff
@@ -0,0 +1,109 @@
+#import <CoreGraphics/CoreGraphics.h>
+
+#import "MGLFoundation.h"
+#import "MGLStyleValue.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+/** Options to specify extruded geometries are lit relative to the map or viewport. */
+typedef NS_ENUM(NSUInteger, MGLLightAnchor) {
+    /** The position of the light source is aligned to the rotation of the map. */
+    MGLLightAnchorMap,
+    /** The position of the light source is aligned to the rotation of the viewport. */
+    MGLLightAnchorViewport
+};
+
+/**
+ A structure containing information about the position of the light source
+ relative to lit geometries.
+ */
+typedef struct MGLLightPosition {
+    /** Distance from the center of the base of an object to its light. */
+    CGFloat radial;
+    /** Position of the light relative to 0° (0° when `lightAnchor` is set to viewport corresponds 
+     to the top of the viewport, or 0° when `lightAnchor` is set to map corresponds to due north, 
+     and degrees proceed clockwise). */
+    CGFloat azimuthal;
+    /** Indicates the height of the light (from 0°, directly above, to 180°, directly below). */
+    CGFloat polar;
+} MGLLightPosition;
```

Comment:
As tail work, maybe we could consider naming this struct MGLSphericalPosition, so it emphasizes the nature of the value instead of how it's used.